apps/stable_diffusion/src/models/model_wrappers.py

```diff
@@ -67,6 +73,70 @@ def check_compilation(model, model_name):
         )
 
 
+def shark_compile_after_ir(
+    module_name,
+    device,
+    vmfb_path,
+    precision,
+    ir_path=None,
+):
+    if ir_path:
+        print(f"[DEBUG] mlir found at {ir_path.absolute()}")
+
+    module = SharkInference(
+        mlir_module=ir_path,
+        device=device,
+        mlir_dialect="tm_tensor",
+    )
+    print(f"Will get extra flag for {module_name} and precision = {precision}")
+    path = module.save_module(
+        vmfb_path.parent.absolute(),
+        vmfb_path.stem,
+        extra_args=get_opt_flags(module_name, precision=precision),
+    )
+    print(f"Saved {module_name} vmfb at {path}")
+    module.load_module(path)
+    return module
+
+
+def process_vmfb_ir_sdxl(extended_model_name, model_name, device, precision):
+    name_split = extended_model_name.split("_")
+    if "vae" in model_name:
+        name_split[5] = "fp32"
+    extended_model_name_for_vmfb = "_".join(name_split)
```

Comment:
This currently appends the entire device URI to the end of the filename, resulting in filenames like "unet_1_77_1024_1024_fp16_stable-diffusion-xl-base-1_vulkan_00000000_0c00_0000_0000_000000000000.vmfb"

Can we make this conform to the naming conventions for non-SDXL models (without the device URI)? e.g.
"unet_1_64_512_512_fp16_stable-diffusion-2-1-base_vulkan.vmfb"

Reply:
The naming of the vmfbs can be done like that but it will be wrong (and the non-SDXL models should also be updated to have the same) - a vmfb compiled for vulkan AMD device should be differentiated from vulkan NVIDIA device. Something similar is being done for Llama as well : [vicuna.py from Top-Of-Main branch](https://github.com/nod-ai/SHARK/blob/main/apps/language_models/scripts/vicuna.py#L1332-L1343), so instead of the entire URI I'm now using the `args.iree-vulkan-target-triple` flag to add the distinction.

So, now you'll see the filenames like : `clip_1_77_1024_1024_fp16_stable-diffusion-xl-base-1_vulkan_rdna2_unknown_linux.vmfb`